pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -238,6 +239,40 @@ func (m *managerImpl) IsUnderPIDPressure() bool {
 	return hasNodeCondition(m.nodeConditions, v1.NodePIDPressure)
 }
 
+func calcAccessibleSwap(pods []*v1.Pod, nodeTotalMemory, totalPodsSwapAvailable uint64) (uint64, error) {
+	klog.V(3).InfoS("calculating accessible swap", "node memory capacity", nodeTotalMemory, "swap capacity", totalPodsSwapAvailable)
+
+	accessibleSwap := uint64(0)
+	for _, pod := range pods {
+		if v1qos.GetPodQOS(pod) != v1.PodQOSBurstable {
+			klog.V(3).InfoS("Accessible swap: skipping since is not of burstable QoS", "pod", pod.Name, "qos", v1qos.GetPodQOS(pod))
+			continue
+		}
+
+		podAccessibleSwap := uint64(0)
+		for _, container := range pod.Spec.Containers {
+			if container.Resources.Requests.Memory() == nil {
+				klog.V(3).InfoS("Accessible swap: skipping since has no memory request", "pod", pod.Name, "container", container.Name)
+				continue
+			}
+
+			containerMemRequest := container.Resources.Requests.Memory()
+			containerAccessibleSwap, err := swap.CalcSwapForBurstablePods(containerMemRequest.Value(), int64(nodeTotalMemory), int64(totalPodsSwapAvailable))
+			if err != nil {
+				return 0, err
+			}
+
+			klog.InfoS("Accessible swap", "pod", pod.Name, "container", container.Name, "container request", containerMemRequest.String(), "accessible swap", containerAccessibleSwap)
+			podAccessibleSwap += uint64(containerAccessibleSwap)
+		}
+		klog.V(3).InfoS("pod accessible swap", "name", pod.Name, "accessible swap", podAccessibleSwap)
+		accessibleSwap += podAccessibleSwap
+	}
+
+	klog.InfoS("Node's accessible swap", "number of pods", len(pods), "accessible swap", accessibleSwap)
```

Comment:
I don't love V(3) in this function. Should we make this higher?

Having lots of logs could remove important events from logs since they would get rotated in a journal.

Reply:
V(5) is probably good.

Reply:
Done :)